### PR TITLE
Fix/vpn server offline error

### LINKED ISSUE #1181: Should get `vpn-server not available` error when connecting to a visor without a active server
**Describe the bug**
Currently, if we try to connect a vpn-client to a VPN server that is not started we do not get a proper reason for not being able to connect.

**Environment information:**

- OS: e.g. Linux
- Platform: Linux 5.13.0-40-generic x86_64

**Steps to Reproduce**

1. Start visor A
2. Start Visor B and connect to the visor A using the vpn-client

**Actual behavior**
visor A
```
[2022-05-09T10:23:39Z] INFO [router]: handling setup request: setupPK(0324579f003e6b4048bae2def4365e634d8e0e3054a20fc7af49daf2a179658557)
ROUTING TABLE CONTENTS: map[8:FWD(keyRtID:8, nxtRtID:8, nxtTpID:3e1a0d82-6e65-06ac-8b7a-f5d4295422e1, rAddr:02aa6fc7009d31622705cb3f2feaa8fe0da10c2ae081cf52e7ba7672bf097defad:49156, lAddr:037695281fc8b59a94b15abe4093845c2ac67f43a59c5d9bcd6e60a449c7342dd2:44)]
[2022-05-09T10:23:39Z] INFO [router]: Save new Routing Rule with ID 8 FWD(keyRtID:8, nxtRtID:8, nxtTpID:3e1a0d82-6e65-06ac-8b7a-f5d4295422e1, rAddr:02aa6fc7009d31622705cb3f2feaa8fe0da10c2ae081cf52e7ba7672bf097defad:49156, lAddr:037695281fc8b59a94b15abe4093845c2ac67f43a59c5d9bcd6e60a449c7342dd2:44)
ROUTING TABLE CONTENTS: map[7:REV(keyRtID:7, rAddr:037695281fc8b59a94b15abe4093845c2ac67f43a59c5d9bcd6e60a449c7342dd2:44, lAddr:02aa6fc7009d31622705cb3f2feaa8fe0da10c2ae081cf52e7ba7672bf097defad:49156) 8:FWD(keyRtID:8, nxtRtID:8, nxtTpID:3e1a0d82-6e65-06ac-8b7a-f5d4295422e1, rAddr:02aa6fc7009d31622705cb3f2feaa8fe0da10c2ae081cf52e7ba7672bf097defad:49156, lAddr:037695281fc8b59a94b15abe4093845c2ac67f43a59c5d9bcd6e60a449c7342dd2:44)]
[2022-05-09T10:23:39Z] INFO [router]: Save new Routing Rule with ID 7 REV(keyRtID:7, rAddr:037695281fc8b59a94b15abe4093845c2ac67f43a59c5d9bcd6e60a449c7342dd2:44, lAddr:02aa6fc7009d31622705cb3f2feaa8fe0da10c2ae081cf52e7ba7672bf097defad:49156)
[2022-05-09T10:23:39Z] INFO [router]: Saving route group rules with desc: rAddr:037695281fc8b59a94b15abe4093845c2ac67f43a59c5d9bcd6e60a449c7342dd2:44, lAddr:02aa6fc7009d31622705cb3f2feaa8fe0da10c2ae081cf52e7ba7672bf097defad:49156
[2022-05-09T10:23:39Z] INFO [router]: Creating new route group rule with desc: rAddr:037695281fc8b59a94b15abe4093845c2ac67f43a59c5d9bcd6e60a449c7342dd2:44, lAddr:02aa6fc7009d31622705cb3f2feaa8fe0da10c2ae081cf52e7ba7672bf097defad:49156
[2022-05-09T10:23:39Z] INFO [RouteGroup rAddr:037695281fc8b59a94b15abe4093845c2ac67f43a59c5d9bcd6e60a449c7342dd2:44, lAddr:02aa6fc7009d31622705cb3f2feaa8fe0da10c2ae081cf52e7ba7672bf097defad:49156]: Sent handshake via transport 3e1a0d82-6e65-06ac-8b7a-f5d4295422e1
[2022-05-09T10:23:40Z] INFO serveRouteGroup [launcher]: Accepted route group. _=skynet local=037695281fc8b59a94b15abe4093845c2ac67f43a59c5d9bcd6e60a449c7342dd2:44 remote=02aa6fc7009d31622705cb3f2feaa8fe0da10c2ae081cf52e7ba7672bf097defad:49156
[2022-05-09T10:23:40Z] INFO [RouteGroup rAddr:037695281fc8b59a94b15abe4093845c2ac67f43a59c5d9bcd6e60a449c7342dd2:44, lAddr:02aa6fc7009d31622705cb3f2feaa8fe0da10c2ae081cf52e7ba7672bf097defad:49156]: Got close packet with code 0
[2022-05-09T10:23:40Z] ERROR [launcher]: no listener on port 44 _=skynet
[2022-05-09T10:23:40Z] INFO [RouteGroup rAddr:037695281fc8b59a94b15abe4093845c2ac67f43a59c5d9bcd6e60a449c7342dd2:44, lAddr:02aa6fc7009d31622705cb3f2feaa8fe0da10c2ae081cf52e7ba7672bf097defad:49156]: Remote got closed, stopping keep-alive loop
[2022-05-09T10:23:40Z] INFO [RouteGroup rAddr:037695281fc8b59a94b15abe4093845c2ac67f43a59c5d9bcd6e60a449c7342dd2:44, lAddr:02aa6fc7009d31622705cb3f2feaa8fe0da10c2ae081cf52e7ba7672bf097defad:49156]: Remote got closed, stopping network probe loop
[2022-05-09T10:23:40Z] WARN [router]: Failed to handle transport frame: routing table: rule not found

```
visor B
```
[2022-05-09T15:42:19+05:30] INFO (STDOUT) [proc:vpn-client:891eb0cbb8974b419e92335f919e2519]: Dialed 037695281fc8b59a94b15abe4093845c2ac67f43a59c5d9bcd6e60a449c7342dd2:44
[2022-05-09T15:42:19+05:30] ERROR [RouteGroup rAddr:02aa6fc7009d31622705cb3f2feaa8fe0da10c2ae081cf52e7ba7672bf097defad:49155, lAddr:037695281fc8b59a94b15abe4093845c2ac67f43a59c5d9bcd6e60a449c7342dd2:44]: error updating activity of rule 5 error="rule of id 5 not found"
[2022-05-09T15:42:19+05:30] INFO (STDOUT) [proc:vpn-client:891eb0cbb8974b419e92335f919e2519]: Sending client hello: {[192.168.0.116 175.0.0.1 172.17.0.1 174.0.0.1 192.168.0.1] }
[2022-05-09T15:42:19+05:30] WARN [proc:vpn-client:891eb0cbb8974b419e92335f919e2519]: Received unexpected error when reading from server. error="EOF"
[2022-05-09T15:42:19+05:30] INFO [proc:vpn-client:891eb0cbb8974b419e92335f919e2519]: Request processed. _elapsed="10.35µs" _method="CloseConn" _received="3:42PM" input=0xc0029c26fa
[2022-05-09T15:42:19+05:30] INFO [proc:vpn-client:891eb0cbb8974b419e92335f919e2519]: Request processed. _elapsed="3.998µs" _method="SetConnectionDuration" _received="3:42PM" input=0
[2022-05-09T15:42:19+05:30] INFO [proc:vpn-client:891eb0cbb8974b419e92335f919e2519]: Request processed. _elapsed="4.879µs" _method="SetDetailedStatus" _received="3:42PM" input=0xc002f6ded0
[2022-05-09T15:42:19+05:30] INFO [proc:vpn-client:891eb0cbb8974b419e92335f919e2519]: Request processed. _elapsed="4.198µs" _method="SetError" _received="3:42PM" input=0xc000518bd0
[2022-05-09T15:42:19+05:30] INFO (STDOUT) [proc:vpn-client:891eb0cbb8974b419e92335f919e2519]: error during client/server handshake: error reading server hello: error reading data: EOFerror serving app conn: error reading server hello: error reading data: EOF
[2022-05-09T15:42:19+05:30] INFO (STDOUT) [proc:vpn-client:891eb0cbb8974b419e92335f919e2519]: Connection broke, reconnecting...
2022/05/09 15:42:19 [erson-69/IpDX2Inhr2-000009] "GET http://localhost:8000/api/visors/02aa6fc7009d31622705cb3f2feaa8fe0da10c2ae081cf52e7ba7672bf097defad/summary HTTP/1.1" from [::1]:42532 - 200 1637B in 1.668384ms
[2022-05-09T15:42:19+05:30] INFO (STDERR) [proc:vpn-client:891eb0cbb8974b419e92335f919e2519]: time="2022-05-09T15:42:19+05:30" level=warning msg=Retrying... current_backoff=2.197s error="dialServeConn: error reading server hello: error reading data: EOF" func=retrier
[2022-05-09T15:42:19+05:30] INFO [proc:vpn-client:891eb0cbb8974b419e92335f919e2519]: Request processed. _elapsed="8.376µs" _method="SetDetailedStatus" _received="3:42PM" input=0xc0032d05c0
[2022-05-09T15:42:19+05:30] INFO [proc:vpn-client:891eb0cbb8974b419e92335f919e2519]: Request processed. _elapsed="6.983µs" _method="SetConnectionDuration" _received="3:42PM" input=0
[2022-05-09T15:42:19+05:30] INFO (STDOUT) [proc:vpn-client:891eb0cbb8974b419e92335f919e2519]: Removing direct route to 172.104.38.100
[2022-05-09T15:42:19+05:30] INFO (STDERR) [proc:vpn-client:891eb0cbb8974b419e92335f919e2519]: time="2022-05-09T15:42:19+05:30" level=info msg="Closing TUN"
[2022-05-09T15:42:19+05:30] INFO (STDOUT) [proc:vpn-client:891eb0cbb8974b419e92335f919e2519]: Removing direct route to 172.104.188.139
[2022-05-09T15:42:19+05:30] INFO (STDERR) [proc:vpn-client:891eb0cbb8974b419e92335f919e2519]: 2022/05/09 15:42:19 rpc.Serve: accept:accept tcp 127.0.0.1:42345: use of closed network connection
[2022-05-09T15:42:19+05:30] INFO (STDOUT) [proc:vpn-client:891eb0cbb8974b419e92335f919e2519]: Removing direct route to 139.162.54.63
[2022-05-09T15:42:19+05:30] INFO (STDOUT) [proc:vpn-client:891eb0cbb8974b419e92335f919e2519]: Removing direct route to 139.162.141.220
[2022-05-09T15:42:19+05:30] INFO (STDOUT) [proc:vpn-client:891eb0cbb8974b419e92335f919e2519]: Removing direct route to 172.104.59.235
[2022-05-09T15:42:19+05:30] INFO (STDOUT) [proc:vpn-client:891eb0cbb8974b419e92335f919e2519]: Failed to serve VPN: failed to connect to the server: maximum retries attempted without success
```

### PR DESCRIPTION
**Pull request by ersonp**
Did you run `make format && make check`?
yes

Fixes #1181

 Changes:	
- Added new packet `ErrorPacket`
- Added new error in vpn-client whitelist

How to test this PR:
1. Start visor A
2. Start visor B
3. From visor A use vpn-client to connect to visor B's pk
4. Check `no listener on port 44, vpn-server offline` error on UI

### CODE REVIEW

**Review by mrpalide — APPROVED** [review]

**Comment by ersonp** [comment]
We will have to update `vpn-monitor` after merging this.

File: pkg/router/router.go [hunk 1]
```diff
@@ -171,7 +171,7 @@ type router struct {
 }
 
 // New constructs a new Router.
-func New(dmsgC *dmsg.Client, config *Config, routeSetupHooks []RouteSetupHook) (Router, error) {
+func New(dmsgC *dmsg.Client, config *Config, routeSetupHooks []RouteSetupHook, logger *logging.Logger) (Router, error) {
```
Comment (jdknives) [inline pkg/router/router.go]: Why are we passing logger explicitly now?

File: pkg/router/router.go [hunk 2]
```diff
@@ -733,6 +735,58 @@ func (r *router) handleKeepAlivePacket(ctx context.Context, packet routing.Packe
...
+		}
+
+		// in this case we have already initialized nrg and may use it straightforward
+		log.Tracef("Got new remote packet with size %d and route ID %d. Using rule: %s",
+			len(packet.Payload()), packet.RouteID(), rule)
+
+		return nrg.handlePacket(packet)
+	}
+
+	// we don't have nrg for this packet. it's either handshake message or
```
Comment (jdknives) [inline pkg/router/router.go]: How can this be a handshake message if we handle an error packet here?

**Comment by jdknives** [comment]
[2022-07-28T22:15:13+02:00] WARN [proc:vpn-client:2a61b69a52ab4ef882498dac42d3eb9c]: Received unexpected error when reading from server. error="EOF" 

We also still log this despite the error being displayed correctly.

**Comment by ersonp** [comment]
> [2022-07-28T22:15:13+02:00] WARN [proc:vpn-client:2a61b69a52ab4ef882498dac42d3eb9c]: Received unexpected error when reading from server. error="EOF"
> 
> We also still log this despite the error being displayed correctly.

This warn log is from the `Read` method of RPC server which is reading on a closed connection. And `EOF` means the conn has been gracefully closed. https://github.com/skycoin/skywire/pull/1325 This PR contains a dirty fix for it where we ignore the warning if the error is `EOF`.

**Pull request merged**